community/ndp/messaging-v1/src/main/java/org/neo4j/ndp/messaging/v1/PackStreamMessageFormatV1.java

```diff
@@ -94,30 +95,22 @@ public int version()
 
     public static class Writer implements MessageFormat.Writer
     {
-        private final PackStream.Packer packer;
-        private final Runnable onMessageComplete;
-
-        public Writer()
+        public static final Runnable NO_OP = new Runnable()
         {
-            this( new BufferedChannelOutput( 8192 ), new Runnable()
+            @Override
+            public void run()
             {
-                @Override
-                public void run()
-                {
-                    // no-op
-                }
-            } );
-        }
+                // no-op
+            }
+        };
+
+        private final PackStream.Packer packer;
+        private final Runnable onMessageComplete;
 
         /**
-         * @param output interface to write messages to
+         * @param packer serializer to output channel
          * @param onMessageComplete invoked for each message, after it's done writing to the output
          */
-        public Writer( PackOutput output, Runnable onMessageComplete )
```

Comment:
Why remove this constructor? I would suggest to keep this constructor to hide packer from outside callers so that we have a structure like 
`data want to send -> writer(packer) -> chunked output stream -> socket output stream`